packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -1010,69 +1010,71 @@ function* relevantRemovedRootsFromFields(
 }
 
 /**
- * Adds any builds missing from the provided change that are relevant to the change.
+ * Adds any refreshers missing from the provided change that are relevant to the change.
  * This function enforces that all relevant removed roots have a corresponding build.
  *
  * @param change - The change with potentially missing builds. Not mutated by this function.
  * @param getDetachedNode - The function to retrieve a tree chunk from the corresponding detached node id.
  * @param removedRoots - The set of removed roots that should be in memory for the given change to be applied.
  * Can be retrieved by calling {@link relevantRemovedRoots}.
  */
-export function addMissingBuilds(
+export function addMissingRefreshers(
 	change: TaggedChange<ModularChangeset>,
 	getDetachedNode: (id: DeltaDetachedNodeId) => TreeChunk | undefined,
 	removedRoots: Iterable<DeltaDetachedNodeId>,
 ): ModularChangeset {
-	const builds: ChangeAtomIdMap<TreeChunk> = new Map();
+	// todo existing refreshers are not copied over as 7251 will remove all existing refreshers anyways
+	const refreshers: ChangeAtomIdMap<TreeChunk> = new Map();
 
 	for (const root of removedRoots) {
-		const node = getDetachedNode(root);
-
-		// if the detached node could not be found, it should exist in the original builds map
-		if (node === undefined) {
-			assert(change.change.builds !== undefined, "detached node should exist");
+		if (change.change.builds !== undefined) {
+			// if the root exists in the original builds map, it does not need to be added as a refresher
 			const original = tryGetFromNestedMap(change.change.builds, root.major, root.minor);
-			assert(original !== undefined, "detached node should exist");
-		} else {
-			setInNestedMap(builds, root.major, root.minor, node);
+			if (original !== undefined) {
+				continue;
+			}
 		}
-	}
 
-	if (builds.size === 0) {
-		return change.change;
+		const node = getDetachedNode(root);
+		assert(node !== undefined, "detached node should exist");
+		setInNestedMap(refreshers, root.major, root.minor, node);
 	}
 
-	if (change.change.builds !== undefined) {
-		populateNestedMap(change.change.builds, builds, true);
+	if (refreshers.size === 0) {
+		return change.change;
 	}
 
-	const { fieldChanges, maxId, revisions, constraintViolationCount, destroys } = change.change;
+	const { fieldChanges, maxId, revisions, constraintViolationCount, builds, destroys } =
+		change.change;
 	return makeModularChangeset(
 		fieldChanges,
 		maxId,
 		revisions,
 		constraintViolationCount,
 		builds,
 		destroys,
+		refreshers,
 	);
 }
 
 /**
- * Removes any builds from the provided change that are not relevant to the change.
- * Calls {@link relevantRemovedRoots} to determine which builds are relevant.
+ * Removes any refreshers from the provided change that are not relevant to the change.
+ * Calls {@link relevantRemovedRoots} to determine which refreshers are relevant.
```

Comment:
We should have removed this line in the last PR. This function is going away in soon so not a big deal.